cartoframes/data/enrichment/polygons_enrichment.py

```diff
@@ -1,83 +1,51 @@
-import uuid
-
-from ..clients import bigquery_client
-from ...auth import get_default_credentials
-from .enrichment_utils import copy_data_and_generate_enrichment_id, process_filters, get_tables_and_variables
-
-
-_ENRICHMENT_ID = 'enrichment_id'
-_WORKING_PROJECT = 'carto-do-customers'
+from .enrichment_service import enrich
 
 # TODO: process column name in metadata, remove spaces and points
 
 
 def enrich_polygons(data, variables, agg_operators, data_geom_column='geometry', filters=dict(), credentials=None):
 
-    credentials = credentials or get_default_credentials()
-    bq_client = bigquery_client.BigQueryClient(credentials)
-
-    user_dataset = credentials.username.replace('-', '_')
-
-    data_copy = copy_data_and_generate_enrichment_id(data, _ENRICHMENT_ID, data_geom_column)
-
-    data_geometry_id_copy = data_copy[[data_geom_column, _ENRICHMENT_ID]]
-    schema = {data_geom_column: 'GEOGRAPHY', _ENRICHMENT_ID: 'INTEGER'}
-
-    id_tablename = uuid.uuid4().hex
-    data_tablename = 'temp_{id}'.format(id=id_tablename)
-
-    bq_client.upload_dataframe(data_geometry_id_copy, schema, data_tablename,
-                               project=_WORKING_PROJECT, dataset=user_dataset, ttl_days=1)
-
-    table_data_enrichment, table_geo_enrichment, variables_list = get_tables_and_variables(variables)
-
-    filters_str = process_filters(filters)
-
-    sql = __prepare_sql(_ENRICHMENT_ID, variables_list, agg_operators, table_data_enrichment,
-                        table_geo_enrichment, user_dataset, _WORKING_PROJECT,
-                        user_dataset, data_tablename, data_geom_column, filters_str)
-
-    data_geometry_id_enriched = bq_client.query(sql).to_dataframe()
-
-    data_copy = data_copy.merge(data_geometry_id_enriched, on=_ENRICHMENT_ID, how='left')\
-        .drop(_ENRICHMENT_ID, axis=1)
+    data_enriched = enrich(_prepare_sql, data=data, variables=variables, agg_operators=agg_operators,
+                           data_geom_column=data_geom_column, filters=filters, credentials=credentials)
 
-    return data_copy
+    return data_enriched
 
 
-def __prepare_sql(enrichment_id, variables, agg_operators, enrichment_table, enrichment_geo_table, user_dataset,
-                  working_project, working_dataset, data_table, data_geom_column, filters):
+def _prepare_sql(enrichment_id, filters_processed, variables_processed, enrichment_table,
+                 enrichment_geo_table, user_dataset, working_project, data_table, **kwargs):
 
     grouper = 'group by data_table.{enrichment_id}'.format(enrichment_id=enrichment_id)
+    agg_operators = None
```

Comment:
We have something wrong here @alejandrohall related with `agg_operators`. I have added here to ensure the following `elif` which is basically checking something that is defined in the previous `if` (so `agg_operators` will always be undefined in the `elif`)

Reply:
Yes, I fixed it!